### PR TITLE
Adding GlobalAvgPool1D, 2D, 3D Test

### PR DESCRIPTION
**Pull request by Shashi456**
Some tests for #77

### CODE REVIEW

File: Tests/DeepLearningTests/LayerTests.swift [hunk 1]
```diff
@@ -42,7 +42,15 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(round(output), expected)
     }
 
+    func testGlobalAvgPool1D() {
+        let layer = GlobalAveragePooling1D<Float>()
+        let input = Tensor<Float>([[0, 1, 2, 3, 4], [10, 11, 12, 13, 14]])
+        let output = layer.inferring(from : input)
```
Comment (rxwei) [inline Tests/DeepLearningTests/LayerTests.swift]: Remove the whitespace before the colon.

File: Tests/DeepLearningTests/LayerTests.swift [hunk 2]
```diff
@@ -42,7 +42,15 @@ final class LayerTests: XCTestCase {
...
+        let expected = Tensor<Float>([2, 10])
+        XCTAssertEqual(round(output, expected))
+    }
+
     static var allTests = [
-        ("testConv1D", testConv1D), ("testMaxPool1D", testMaxPool1D), ("testAvgPool1D", testAvgPool1D)
+        ("testConv1D", testConv1D), ("testMaxPool1D", testMaxPool1D), ("testAvgPool1D", testAvgPool1D), ("testGlobalAvgPool1D", testGlobalAvgPool1D)
```
Comment (rxwei) [inline Tests/DeepLearningTests/LayerTests.swift]: Break this into multiple lines.

**Comment by Shashi456** [comment]
@rxwei yeah I will add tests for the rest of the global average pooling layers too, I was just checking if I had the idea of the input and output right.

**Comment by Shashi456** [comment]
@rxwei so the GlobalAveragePooling1D function is throwing an error because of the `.reshaped(to: [input.shape[0], input.shape[2]]` , it makes sense, since in the test I initialized `let input = Tensor<Float>([[0, 1, 2, 3, 4], [10, 11, 12, 13, 14]])` whose shape is [2,5]. It doesnt have a shape[2]. How do we rewrite that?

**Comment by Shashi456** [comment]
@rxwei I think this can be merged now.

**Comment by rxwei** [comment]
Thanks!

**Pull request merged**